File: content/examples/other/leetcode/problems/product-of-array-except-self/example-2/Solution.java
import java.util.*;

public class Solution {
  public static int[] productExceptSelf(int[] nums) {
    int n = nums.length;
    int[] out = new int[n];
    int prefix = 1;
    for (int i = 0; i < n; i++) {
      out[i] = prefix;
      prefix *= nums[i];
    }
    int suffix = 1;
    for (int i = n - 1; i >= 0; i--) {
      out[i] *= suffix;
      suffix *= nums[i];
    }
    return out;
  }

  public static void main(String[] args) {
    System.out.println(Arrays.toString(productExceptSelf(new int[]{1,2,3,4})));
  }
}

File: content/examples/other/leetcode/problems/product-of-array-except-self/example-2/solution.js
function productExceptSelf(nums) {
  const n = nums.length;
  const out = new Array(n);
  let prefix = 1;
  for (let i=0;i<n;i+=1) {
    out[i] = prefix;
    prefix *= nums[i];
  }
  let suffix = 1;
  for (let i=n-1;i>=0;i-=1) {
    out[i] *= suffix;
    suffix *= nums[i];
  }
  return out;
}

if (require.main===module) {
  console.log(productExceptSelf([1,2,3,4]));
}

module.exports = { productExceptSelf };
